qa/tasks/caasp.py

```diff
@@ -0,0 +1,137 @@
+'''
+Task that deploys a CAASP cluster on all the nodes
+Linter:
+    flake8 --max-line-length=100
+'''
+import logging
+import os
+import subprocess
+from util import remote_exec
+from teuthology.exceptions import ConfigError
+from teuthology.misc import (
+    delete_file,
+    move_file,
+    sh,
+    sudo_write_file,
+    write_file,
+    copy_file,
+    all_roles_of_type
+    )
+from teuthology.orchestra import run
+from teuthology.task import Task
+from util import (
+    get_remote_for_role,
+    remote_exec
+    )
+log = logging.getLogger(__name__)
+
+
+class Caasp(Task):
+    """
+    Deploy a Salt cluster on all remotes (test nodes).
+    This task assumes all relevant Salt packages (salt, salt-master,
+    salt-minion, salt-api, python-salt, etc. - whatever they may be called for
+    the OS in question) are already installed. This should be done using the
+    install task.
+    One, and only one, of the machines must have a role corresponding to the
+    value of the variable salt.sm.master_role (see salt_manager.py). This node
+    is referred to as the "Salt Master", or the "master node".
+    The task starts the Salt Master daemon on the master node, and Salt Minion
+    daemons on all the nodes (including the master node), and ensures that the
+    minions are properly linked to the master. Finally, it tries to ping all
+    the minions from the Salt Master.
+    :param ctx: the argparse.Namespace object
+    :param config: the config dict
+    """
+
+    def __init__(self, ctx, config):
+        super(Caasp, self).__init__(ctx, config)
+        log.debug("beginning of constructor method")
+        self.ctx['roles'] = self.ctx.config['roles']
+        self.log = log
+        self.remotes = self.cluster.remotes
+        self.mgmt_remote = get_remote_for_role(self.ctx, "skuba_mgmt_host.0")
+        self.ssh_priv = 'tempkey.rsa'
+        self.ssh_pub = 'tempkey.rsa.pub'
+        self.set_agent = "eval `ssh-agent` && ssh-add ~/.ssh/%s && " % self.ssh_priv
+
+    def __ssh_setup(self):
+        log.debug("Executing SSH setup")
+        self.__ssh_gen_key()
+        self.__ssh_copy_priv()
+        self.__ssh_copy_pub_to_caasp()
+
+    def __ssh_gen_key(self):
+        if os.path.isfile(self.ssh_priv):
+            os.remove(self.ssh_priv)
```

Comment:
This is not the teuthology main key.
We are generating a new key for each job, which in turn avoid interacting with machines not supposed to.

Reply:
Why should we delete it at all, my suggestion is not to store this keys to worker at all, since it is the cluster internal usage key. Just create the keys on remote manager, copy them to all remotes and forget about them.